src/Compilers/VisualBasic/Test/Emit/Emit/DeterministicTests.vb

```diff
@@ -132,6 +133,67 @@ Partial.c = 3
         End Sub
 
         <Fact>
+        <WorkItem(11990, "https://github.com/dotnet/roslyn/issues/11990")>
+        Public Sub ForwardedTypesAreEmmittedInADeterministicOrder()
+            Const generatedTypes = 100
+
+            ' VBC doesn't recognize type forwards in VB source code. Because of that,
+            ' this test generates types as a C# library (1), then generates a C# netmodule (2) that
+            ' contains the type forwards that forwards to reference (1), then generates an empty VB
+            ' library (3) that contains (2) and references (1).
+
+            Dim forwardedToCode = New StringBuilder()
+            forwardedToCode.AppendLine("namespace ForwardedToNamespace {")
+            For i = 1 To generatedTypes
+                forwardedToCode.AppendLine($"public class T{i} {{}}")
+            Next
+            forwardedToCode.AppendLine("}")
+
+            Dim forwardedToCompilation = CreateCSharpCompilation(
+                forwardedToCode.ToString(),
+                compilationOptions:=New CSharp.CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary))
+
+            Dim compileAndGetExportedTypes As Func(Of List(Of String)) =
+            Function()
+                Dim forwardingCode = New StringBuilder()
+                forwardingCode.AppendLine("using System.Runtime.CompilerServices;")
+                For i = 1 To generatedTypes
+                    forwardingCode.AppendLine($"[assembly: TypeForwardedTo(typeof(ForwardedToNamespace.T{i}))]")
+                Next
+
+                Dim forwardingNetModule = CreateCSharpCompilation(
+                    "ForwardingAssembly",
+                    forwardingCode.ToString(),
+                    compilationOptions:=New CSharp.CSharpCompilationOptions(OutputKind.NetModule),
+                    referencedAssemblies:={MscorlibRef, SystemRef, forwardedToCompilation.EmitToImageReference()})
+
+                Dim forwardingCompilation = CreateCompilationWithMscorlib(
+                    assemblyName:="ForwardingAssembly",
+                    source:=String.Empty,
+                    options:=New VisualBasicCompilationOptions(OutputKind.DynamicallyLinkedLibrary),
+                    references:={forwardingNetModule.EmitToImageReference(), forwardedToCompilation.EmitToImageReference()})
```

Comment:
It looks like this issue is not addressed.

---
In reply to: [95459717](https://github.com/dotnet/roslyn/pull/16303#discussion_r95459717) [](ancestors = 95459717)